src/java.base/share/classes/java/util/stream/AbstractPipeline.java

```diff
@@ -85,7 +85,7 @@ abstract class AbstractPipeline<E_IN, E_OUT, S extends BaseStream<E_OUT, S>>
      * The "upstream" pipeline, or null if this is the source stage.
      */
     @SuppressWarnings("rawtypes")
-    private final AbstractPipeline previousStage;
+    protected final AbstractPipeline previousStage;
```

Comment:
Making this accessible in subclasses allows to avoid having to store this reference twice, and since this value is passed in during ctor it doesn't expose something previously hidden.

Reply:
Since stream facilities are package-private, we can just use no access modifier and remove all new `protected` access modifier (on methods, fields, constructors) in this PR.

Reply:
@liach We could do that, yet I'm not sure that is strictly an improvement. What would be the benefit in doing so specifically for this PR?